examples/go/serverless/README.md

```diff
@@ -0,0 +1,93 @@
+# Example Serverless App
+
+This app is similar to [Hello, World!](../hello-world) app, but is buildable
+for, and deployable to AWS Lambda and OpenFaaS, in addition to being runnable as
+an HTTP server - for debugging, or otherwise.
+
+It is also possible, but is not illustrated in this example, to package go apps
+as Mattermost Plugins so that they can run on the Mattermost server itself,
+without another hosting platform. See the work-in-progress [Google Calendar
+App](https://github.com/mattermost/mattermost-app-gcal) as an example.
+
+
+
+## What's in the App?
+
+This app contains:
+
+- [manifest](./manifest.json). In addition to what's in [Hello, World!](../hello-world),
+  this app declares several `Deploy` modes: `http`, `aws_lambda`, and
+  `open_faas`. Note that the manifest is packaged as a separate file, to access
+  from [appsctl](https://developers.mattermost.com/integrate/apps/deploy/) for
+  deployment.
+- [Makefile](./Makefile) is used to build the bundles to be deployed with
+  [appsctl](https://developers.mattermost.com/integrate/apps/deploy/).
+- `./function` contains the "core" of the app, everything except the static
+  files, and platform-specific `main` packages is there. Note that because
+  OpenFaaS deployment presently builds the app from its own template, the
+  `function` package has its own `go.mod`.
+- `./aws`, `./http` - main "stubs" for the respective platforms.
+- `./openfaas` - support files to build the OpenFaaS-deployable bundle.
+- `./static` - the app's icon.
+
+## Prerequisites
+
+See [Hello, World!](../hello-world) for the steps to set up a minimal dev environment.
+
+You also will need [appsctl](https://developers.mattermost.com/integrate/apps/deploy/)
+installed, see the link for instructions.
+
+## Run as HTTP
+
+To run the app as a localhost HTTP server,
+```bash
+cd ./examples/go/serverless
+make run
+```
+
+It will print out the exact `/apps install` command to install it onto Mattermost.
+
+## Deploy and run on AWS
+
+If you have an AWS Account, and would like to deploy and run the app as a lambda function,
+
+```bash
+cd ./examples/go/serverless
+make dist-aws
+```
+
+This will create a `./dist/bundle-aws.zip` file containg the AWS-deployable app bundle. Follow the steps in [Admin's Guide](https://developers.mattermost.com/integrate/apps/deploy/deploy-aws) to setup your `appsctl` environment and initialize the AWS resources with `appsctl aws init` command. Then execute
```

Comment:
The linked guide is ~7 months old, have there been any reports of issues with it or planned changes? Just wanted to get a scope on that as part of this. If no one has tested the instructions in a while, it would be nice to run through that guide then this new one to make sure it all works together.

Reply:
Just closing the loop, I finished the above task over on [this Jira ticket](https://mattermost.atlassian.net/browse/MM-43335).